### PR TITLE
JIT: Add a stress mode that poisons implicit byrefs

### PR DESCRIPTION
**Pull request by jakobbotsch**
This stress mode poisons all implicit byrefs before returns from the method. GC pointers are nulled out and other parts of the structs are filled with 0xcd bytes.

This should help expose incorrectly elided copies in the recently added last-use copy elision optimization.

### CODE REVIEW

File: src/coreclr/jit/importer.cpp [hunk 1]
```diff
@@ -11204,6 +11210,87 @@ bool Compiler::impReturnInstruction(int prefixFlags, OPCODE& opcode)
...
+            if (start > 0)
+            {
+                addr = gtNewLclFldAddrNode(lclNum, start, TYP_BYREF);
+            }
+            else
+            {
+                addr = gtNewLclVarAddrNode(lclNum, TYP_BYREF);
+            }
+
+            GenTree* blk = new (this, GT_BLK) GenTreeBlk(GT_BLK, TYP_STRUCT, addr, typGetBlkLayout(count));
```
[the remark this answers is not in the document]
Comment (jakobbotsch) [reply src/coreclr/jit/importer.cpp]: Thanks, addressed I think. Can you take another look?

File: src/coreclr/jit/importer.cpp [hunk 2]
```diff
@@ -11204,6 +11210,83 @@ bool Compiler::impReturnInstruction(int prefixFlags, OPCODE& opcode)
     return true;
 }
 
+#ifdef DEBUG
+//------------------------------------------------------------------------
+// impPoisonImplicitByrefsBeforeReturn:
+//   Spill the stack and insert IR that poisons all implicit byrefs.
+//
+void Compiler::impPoisonImplicitByrefsBeforeReturn()
```
Comment (SingleAccretion) [inline src/coreclr/jit/importer.cpp:11218]: Nit: a little comment for why do we want to do this?

**Review by SingleAccretion — APPROVED** [review]

File: src/coreclr/jit/importer.cpp (hunk 2 above)
Comment (jakobbotsch) [reply src/coreclr/jit/importer.cpp]: Done!

**Comment by jakobbotsch** [comment]
The failures are known or will be fixed by #80734.

cc @dotnet/jit-contrib PTAL @AndyAyersMS

**Review by AndyAyersMS — APPROVED** [review]
Thanks for adding this, it looks like a good safeguard.

**Comment by jakobbotsch** [comment]
Failures are #80666 and #80773

**Pull request merged**